libndt.hpp

```diff
@@ -1922,10 +1926,14 @@ bool Client::ndt7_download() noexcept {
       // string is problematic because our size is 64 bit while size_t is 32
       // bit on the platfrom. That said, it's unlikely that the we'll get a
       // measurement that big, so the check to make sure the casting is okay
-      // is not going to be a real probem, it's just a theoric issue.
+      // is not going to be a real problem, it's just a theoric issue.
       if (count <= SIZE_MAX) {
         std::string sinfo{(const char *)buff.get(), (size_t)count};
-        on_result("ndt7", "download", std::move(sinfo));
+        if (settings_.batch_mode) {
+          std::cout << sinfo;
+        } else {
+          on_result("ndt7", "download", std::move(sinfo));
+        }
```

Comment:
Can you maybe override `on_result` in the main client and move this logic there?